### PR TITLE
Don't use a silent context silent to ensure PIN prompts appear if CNG/CAPI requires it

### PR DESCRIPTION
**Pull request by clairernovotny**
The current overload of `ComputeSignature` changed meaning between .NET Framework and .NET Core. In .NET Core, the context is silent by default, which won't enable PIN prompts for smart cards/usb tokens that require it. `false` must be passed in to enable that if applicable.

Will block code signing from working from most USB thumb drives if not set.

@heng-liu @dtivel

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs
Comment (zivkan) [inline src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:139]: This should only be done when restore was done in `--interactive` mode, right? Otherwise it could cause CI machines to hang if a cert PIN popup comes up.

Comment (zivkan) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:139]: I didn't think long enough about my message. It's not getting signed in restore, it's a separate command. If the sign command doesn't have a `--interactive` command, I think it should be added to reduce risk on CI machines. If sign does have an interactive mode, I think it should be used here.

Comment (clairernovotny) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:139]: @zivkan agreed, just pointing out that this is existing behavior when on the .NET Framework today. .NET Core flipped the default overload's meaning in this case.

Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:139]: NuGet.exe by default runs in an interactive mode. 
However, there is a [noninteractive switch](https://docs.microsoft.com/en-us/nuget/reference/cli-reference/cli-ref-sign#options) that can be used.

https://github.com/NuGet/NuGet.Client/blob/502505da868f79b1c4fc1704dae6642138adb30c/src/NuGet.Core/NuGet.Commands/SignCommand/CertificateFindOptions.cs#L52 already propagates that information so it just needs to be wired in. 
Taking a quick look some heavier refactoring might be required.

**Review by nkolev92 — CHANGES_REQUESTED** [review]

File: src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs
Comment (clairernovotny) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:139]: @nkolev92 are you asking me to do this bigger refactoring? I agree it should be done, however I'm not able to do that bigger piece. I'm just trying to preserve current behavior.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:139]: Please correct me if I'm missing something, but this change would only be relevant in the .NET Core signing case, which is still a WIP by @dtivel and @heng-liu. 

In that case, wouldn't this be better tracked with the rest of the .NET Core signing work?

Comment (clairernovotny) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:139]: Yes, I wasn't sure the right branch to submit this PR to, it seems like it was this one?

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Packaging/Signing/Authoring/X509SignatureProvider.cs:139]: Oh, I didn't even see the branch, my bad. 

I'll defer to @heng-liu as the branch owner here. 

It might take some time to receive feedback as she's making her way through all the issues we see :)

**Comment by nkolev92** [comment]
I dismissed my review as I did not notice the branch .
My point still does stand, but I'm letting @zkat and @heng-liu handle it as the owners.

**Pull request closed without merging**
